include/mbedtls/check_config.h

```diff
@@ -280,7 +280,8 @@
 #endif
 
 #if defined(MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED) &&                 \
-    ( !defined(MBEDTLS_ECDH_C) || !defined(MBEDTLS_ECDSA_C) ||          \
+    ( !defined(MBEDTLS_ECDH_C) ||                                       \
+      !(defined(MBEDTLS_ECDSA_C) || defined(PSA_HAVE_FULL_ECDSA)) ||     \
       !defined(MBEDTLS_X509_CRT_PARSE_C) )
 #error "MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED defined, but not all prerequisites"
```

Comment:
OK, I think I know what's the issue: 
if in `component_test_psa_crypto_config_accel_ecdsa` I remove
```
    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
    scripts/config.py unset MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
```
everything builds and tests fine (as the test suites using ECDHE_ECDSA are skipped, there's no `MBEDTLS_PK_CAN_ECDSA_SOME`), ssl-server2 and ssl-client2 report that they can use ECDHE-ECDSA ciphersuites, but the handshake fails.
Is it a gap in the dependencies here?

Reply:
Ah, good catch! Indeed now that you mention it the ECDSA part of the dependency should be:
```c
((!defined(MBEDTLS_USE_PSA_CRYPTO) && defined(MBEDTLS_ECDSA_C)) || \
 (defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_HAVE_FULL_ECDSA)))
```
Wow, sorry for missing that and thanks for trying things out and catching it!

I wanted to do a final ECDSA PR to add a ChangeLog entry, I'll fix that there too.

Reply:
Yep, that's the second time that that tricky pair of `MBEDTLS_USE_PSA_CRYPTO` vs `!MBEDTLS_USE_PSA_CRYPTO` catches us :) Thanks!

Reply:
Ok, it took me a while to understand the point here, but I agree with you now :)
@AndrzejKurek thanks for the hint!

@mpg you can assign the new issue to me so I will complete this part

Reply:
Or can I simply create a pull request to fix this part without an associated issue?

Reply:
Ah, sorry, I had already started working on it before I was your comment.

But in general yes, it's OK to create a PR without an issue - we just need to add a size label to the PR and add it to the EPICs board in the correct EPIC. (PRs linked to an issue don't need that as they show on the board attached to their issue.)

Reply:
> Ah, sorry, I had already started working on it before I was your comment.

No problem! Thanks for taking care of it ;)

Reply:
https://github.com/Mbed-TLS/mbedtls/pull/7245